operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -74,7 +74,17 @@ func CreateOrUpdateLokiStack(ctx context.Context, req ctrl.Request, k k8s.Client
 		tenantSecrets   []*manifests.TenantSecrets
 		tenantConfigMap map[string]openshift.TenantData
 	)
-	if flags.EnableGateway && stack.Spec.Tenants != nil {
+	if flags.EnableGateway && stack.Spec.Tenants == nil {
+		statusErr := status.SetDegradedCondition(ctx, k, req,
+			"Invalid tenants configuration - TenantsSpec cannot be nil when gateway flag is enabled",
+			lokiv1beta1.ReasonInvalidTenantsConfiguration,
+		)
+		if statusErr != nil {
+			return kverrors.Wrap(statusErr,
+				"Failed to update lokistack condition with invalid tenants configuration")
+		}
+		return kverrors.New("Invalid tenants configuration - TenantsSpec cannot be nil when gateway flag is enabled")
```

Comment:
From the comment on #5373 it occurred to me that we do not want a retry here as well: the operator is watching the `LokiStack` resources and as long as the resource does not change it will not come to a different conclusion regarding this check. So it does not make sense to retry the loop and potentially have a long backoff when a change to the resource finally happens.

Reply:
This is different use-case from that @xperimental 
In case of lokistack gateway which is managed by us, we need to show users that they are giving invalid configuration and hence we need to run the operator in degraded mode and as soon as they update the configuration, the operator should reconcile it automatically. 
The thing with aws secrets is that it is third party and hence we don't want the operator to reconcile automatically.

Reply:
I don't think we should mix both the use-cases. Even for other validations related to gateway, we do the same thing what I am doing here. Hence it would be consistent.

Reply:
> In case of lokistack gateway which is managed by us, we need to show users that they are giving invalid configuration and hence we need to run the operator in degraded mode and as soon as they update the configuration, the operator should reconcile it automatically.

I agree and I think the experience for users is much better when they do not have to wait for a back-off to clear before the operator reacts on their change again. In the current version the operator is re-trying the reconciliation immediately and because it will come to the same conclusion (as the `LokiStack` has not changed in-between) it will drift towards a long back-off. Once the user fixes the error by adding a tenant configuration to the LokiStack the loop will be triggered again but the operator will not react until the back-off has elapsed.

What I am suggesting is setting the degraded status but not raising an error, in which case the operator will not retry immediately (and so not trigger any back-off). Instead it will trigger the reconciliation on the next event (possibly the user's change of the tenant configuration) where the reaction is not limited by a back-off but proceeds immediately.